### PR TITLE
fix: split c8y JSON over MQTT messages using newline delimiter

### PR DESCRIPTION
**Pull request by Bravo555**
## Proposed changes

#3301 was wrong: using a function meant for splitting Smartrest will mess up JSON over MQTT since that function will unescape \" sequences which we want to do in Smartrest but avoid in JSON over MQTT. This PR undoes this function's rename and uses regular line splitting for JSON over MQTT.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue

- #3297

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by albinsuresh — APPROVED** [review]
LGTM

**Comment by reubenmiller** [comment]
Given the nature of this PR I also verified it end-to-end just to be sure it works as expected.

File: crates/extensions/c8y_mapper_ext/src/converter.rs [hunk 1]
```diff
@@ -634,11 +634,11 @@ impl CumulocityConverter {
     ) -> Result<Vec<MqttMessage>, ConversionError> {
         // JSON over MQTT messages on c8y/devicecontrol/notifications can contain multiple operations in a single MQTT
         // message, so split them
-        let operation_payloads = collect_c8y_messages(message.payload_str()?);
+        let operation_payloads = message.payload_str()?.lines();
```
Comment (didier-wenzek) [inline crates/extensions/c8y_mapper_ext/src/converter.rs:637]: Okay - assuming that Cumulocity emits a batch of messages using JSON lines. A point I fail to find in Cumulocity docs.

**Review by didier-wenzek — APPROVED** [review]
Approved

File: crates/extensions/c8y_mapper_ext/src/converter.rs (hunk 1 above)
Comment (reubenmiller) [reply crates/extensions/c8y_mapper_ext/src/converter.rs:637]: Unfortunately I don't think this is publicly documented.

**Pull request merged**